mvcc/watchable_store.go

```diff
@@ -211,18 +211,20 @@ func (s *watchableStore) watch(key, end []byte, startRev int64, id WatchID, ch c
 
 	cancel := cancelFunc(func() {
 		s.mu.Lock()
-		defer s.mu.Unlock()
+
```

Comment:
change to `if ... { ... } else if ... { ... }` instead of `if ... { ... return} if { ... }` so there's still only one `s.mu.Unlock()` call site?